2to3: add page (#5015)
tldr-pages/tldr: pages/common/2to3.md

# 2to3

> Automated Python 2 to 3 code translation.
> More information: <https://docs.python.org/3/library/2to3.html>.

> - Print changes that would be performed:

Reviewer: ```suggestion
- Display changes that would be performed:
```

> `2to3 {{file.py}}`

Reviewer: ```suggestion
`2to3 {{path/to/file.py}}`
```

> - Convert Python 2 file to Python 3:

Reviewer 2: ```suggestion
- Convert a Python 2 file to Python 3:
```

> `2to3 -w {{file.py}}`

Reviewer: Use long argument format please:
```suggestion
`2to3 --write {{path/to/file.py}}`
```

> - Convert specified Python 2 features to Python 3:

Reviewer 2: ```suggestion
- Convert a specified Python 2 feature to Python 3:
```

> `2to3 -w {{file.py}} -f {{raw_input}} -f {{print}}`

Reviewer: ```suggestion
`2to3 --write {{file.py}} --fix={{raw_input}} --fix={{print}}`
```

> - Convert all features except *raw_input* the specified ones to Python 3:

Reviewer: We use back ticks here:
```suggestion
- Convert all features except `raw_input` the specified ones to Python 3:
```

Author: @navarroaxel I know I saw `*...*` somewhere...

Author: [`pages/common/strings.md`](https://github.com/tldr-pages/tldr/blob/master/pages/common/strings.md)

Author: [`pages/common/supervisorctl.md`](https://github.com/tldr-pages/tldr/blob/master/pages/common/supervisorctl.md)

Author: @navarroaxel `strings` uses `*`, and `supervisorctl` uses `**`.  Should I use the backtick as you've done above, and open a PR to normalize those ones that I've mentioned?

Reviewer: Well... We should improve our standard. Leave the * then.

Author: @navarroaxel should I open a PR to correct those?  I can correct the standard, too :)

Author: @owenvoke WDYT?

Reviewer 3: For the `supervisorctl`, I don't think it needs the bold text (and as it's not part of the standard, it looks weird).

As for the italics in `strings`, I think we could just remove it there too, and replace the "length" with an example number. e.g.

```
- Limit results to strings that are at least 10 characters long:

`strings -n {{10}} {{file}}`
```

Author: >and replace the "length" with an example number

Good idea.  That must be the bit that I was missing.  Thank you!

> `2to3 -x {{raw_input}} -f {{print}} example.py`

Reviewer: Should the 2nd argument be `--nofix`?
```suggestion
`2to3 --nofix={{raw_input}} --fix={{print}} example.py`
```

Author: @navarroaxel sure.  I just copied from the docs page.

> `2to3 --nofix={{raw_input}} --fix={{print}} example.py`

Reviewer: I just want to confirm: should this `--fix={{print}}` be `--nofix={{print}}`?

Author: @navarroaxel Yes.

Author: @navarroaxel Maybe it should be:
```
Convert all features except *raw_input* to Python 3:
```

Reviewer: Yeah, please apply.

Author: @navarroaxel done @ 17badf8d78705416d4880d793d66d97697cb51c1

> - Print list of available convertable features:

Reviewer: ```suggestion
- Display the list of available transformation features:
```

> `2to3 -l`

Reviewer: ```suggestion
`2to3 --list-fixes`
```

- Convert an entire directory:

> `2to3 --output-dir={{python3-version/mycode}} -W -n {{python2-version/mycode}}`

Reviewer: ```suggestion
`2to3 --output-dir={{path/to/code_python3_version}} --write-unchanged-files --nobackups {{path/to/code_python2_version}}`
```
